pkg/deployment/deployment.go

```diff
@@ -69,18 +69,24 @@ func NewBuilder(
 		glog.V(100).Infof("The name of the deployment is empty")
 
 		builder.errorMsg = "deployment 'name' cannot be empty"
+
+		return &builder
```

Comment:
No need for this repeated return. We have validate method which will print that builder has error and return the exact error.

Reply:
We use builder.errorMsg in our unit test. You just need to check that builder has relevant error.

Reply:
I think we've talked about this before for other parts of the code where you could potentially have a blank string for name, namespace and an empty `if len(labels) == 0 {` and the last condition would be the builder.errorMsg that was set.

Reply:
yes and It's OK because code will not work until all errors I'll be fixed.